network/src/protocol.rs

```diff
@@ -92,8 +98,8 @@ impl ServiceProtocol for CKBHandler {
     fn init(&mut self, _control: &mut ServiceContext) {}
     fn connected(&mut self, control: &mut ServiceContext, session: &SessionContext, version: &str) {
         let (peer_id, version) = {
-            // FIXME: version number should be discussed.
-            let parsed_version = version.split('.').last().and_then(|v| v.parse::<u8>().ok());
+            // TODO: version number should be discussed.
+            let parsed_version = version.parse::<u8>().ok();
```

Comment:
How about use https://github.com/steveklabnik/semver ?

Reply:
Make sense to me, but need some API from underlying p2p library for select version.

Reply:
It is over enginery, network protocol should not be such complex.

Reply:
Version select **is** [complex](https://github.com/bitcoin/bitcoin/blob/717fd58c4ba5c64778fa9a8bf6bbf5ae1164df3f/src/net_processing.cpp#L1808-L1936).
It's about you handle it in library or in user defined protocol.